renderdoc/driver/shaders/dxbc/dxbc_debug.cpp

```diff
@@ -573,6 +585,25 @@ bool State::Finished() const
   return dxbc && (done || nextInstruction >= (int)dxbc->GetNumInstructions());
 }
 
+void State::AssignValue(ShaderVariable &dst, uint32_t dstIndex, const ShaderVariable &src,
+                        uint32_t srcIndex)
+{
+  if(src.type == eVar_Float)
+  {
+    float ft = src.value.fv[srcIndex];
+    if(isinf(ft) || isnan(ft))
+      flags |= eShaderDebugStateFlags_GeneratedNanOrInf;
+  }
+  else if(src.type == eVar_Double)
+  {
+    double dt = src.value.dv[srcIndex];
+    if(isinf(dt) || isnan(dt))
+      flags |= eShaderDebugStateFlags_GeneratedNanOrInf;
+  }
```

Comment:
I wonder if the `mov` instruction should be excluded from this somehow. That way at least when an instruction generates an inf, you won't get false positives after that from moving it around, even if you do get a hit every time it's used in a subsequent math operation.

Either that or more complicated, you could look to see if all float inputs are *not* non-normal, and the output *is* non-normal. I'm not sure that can be done just globally without any per-operation markup/knowledge.

Once you've run to the generation of the first nan/inf then things tend to quickly spiral out of control after that regardless, so it's maybe not useful to be able to run to the _second_ generated nan/inf. What do you think? Worth the extra complication to do now, or just something to leave until later if it's desired later?